lib/krikri/parser.rb

```diff
@@ -103,5 +103,40 @@ def get_child_nodes(name)
         raise NotImplementedError, "Can't access property #{name}"
       end
     end
+
+    class ValueArray < Array
+      def values
+        map(&:value)
+      end
+
+      def field(*args)
+        result = self
+        args.each do |name|
+          result = result.get_field(name)
+        end
+        result
+      end
+
+      def select(*)
+        self.class.new(super)
+      end
+
+      def match_attribute(name, other)
```

Comment:
After looking at the [gist of the ESDN mapping](https://gist.github.com/no-reply/a3369d930c7cba223138), I realized I wanted to ask whether `Krikri::Parser::ValueArray#match_attribute` calls can be chained together, e.g., ESDN's `isShownAt` should actually be more like this based on the crosswalk:

```ruby
  isShownAt :class => DPLA::MAP::WebResource do
    uri record.field('mods:location', 'mods:url').match_attribute(:usage, 'primary').match_attribute(:access, 'object in context')
  end
```